File: CS50/pset1/cash/credit.py
number = int(input("enter a credit card number: "))
len_num = len(str(number))
ctr = 0
total = 0
for i in range(len_num):
    ctr += 1
    temp = number % 10
    if ctr % 2 == 0:
        temp1 = temp * 2
        if (temp1 >= 10):
            temp1 = temp1 % 10 + 1
            total += temp1
        else:
            total += temp1
    else:
        total += temp
    number -= temp
    number = int(number / 10)
print(total)

File: CS50/pset1/cash/credit.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

long get_long(void){
    long int num;
    do
    {
        printf("enter a credit card number: ");
        scanf("%ld", &num);
    } while (num < 1 || num > 9999999999999999);
    return num;
}


int main(){
    long int number = get_long();
    int nDigits = floor(log10(labs(number))) + 1;
    printf("digit %d", nDigits);
    int ctr = 0;
    int sum = 0;
    for(int i=0; i<nDigits; i++){
        //check for odd or even
        ctr += 1;
        
        if(ctr % 2 == 0){
            int temp = number % 10;
            temp *= 2;
            if(temp >= 10){
                temp = temp % 10 + 1;
                sum += temp;
            }else{
                sum += temp;
            }
        }else{
            sum += number % 10;
        }
    }
    printf("%d", sum);
    return 0;
}